Tasks/Maven/maven.ps1

```diff
@@ -28,14 +32,37 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
+# If JAVA_HOME is being set by choosing a JDK version find the path to that specified version else use the path given by the user
+$jdkPath = $null
+if($javaHomeSelection -eq 'JDKVersion')
 {
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
+    Write-Verbose "Using JDK version to find and set JAVA_HOME"
+    # If the JDK version is not the deafult set the jdkPath to the new JDK version selected
+    if($jdkVersion -and $jdkVersion -ne "default")
     {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
+        $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
+        if (!$jdkPath) 
+        {
+            throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
```

Comment:
We should localize these messages. Check with Eric on the build team on how to do that.